### PR TITLE
[8] GET /todos/:id

### PR DESCRIPTION
**Pull request by pdany1116**
#### Description

Retrieve a single todo via GET /todos/:id.

## Steps to Test or Reproduce

1. Follow test steps in README.
2. All tests should pass.

### CODE REVIEW

File: features/todo.feature [hunk 1]
```diff
@@ -6,3 +6,8 @@ Feature: Todo API
     Given a todo exists
     When I retrieve the list of todos
     Then I will get a response with a list containing a todo
+
+  Scenario: Get a todo
```
Comment (cazacutudor) [inline features/todo.feature:10]: We should run the integrations specs via CI.

File: app/controllers/todos_controller.rb [hunk 2]
```diff
@@ -6,4 +6,12 @@ def index
 
     render json: todos, status: :ok
   end
+
+  def show
+    todo = Todo.find(params[:id])
+
+    render json: todo, status: :ok
```
Comment (pdany1116) [inline app/controllers/todos_controller.rb:13]: Serialize the todo response. We do not want to expose sensitive data. To be covered in another ticket.

File: features/support/todo_helpers.rb [hunk 3]
```diff
@@ -5,9 +5,14 @@ module TodoHelpers
...
     create(:todo)
+    @last_todo_id = Todo.last.id
   end
 
   def get_todos
     get '/todos'
   end
+
+  def get_todo(id)
+    get "/todos/#{id}"
```
Comment (pdany1116) [inline features/support/todo_helpers.rb]: Use predefined paths by rails.

File: features/todo.feature (hunk 1 above)
Comment (pdany1116) [inline features/todo.feature:10]: Let's add the integration tests in CI.

File: spec/requests/todos_spec.rb [hunk 4]
```diff
@@ -36,4 +36,33 @@
       end
     end
   end
+
```
Comment (cazacutudor) [inline spec/requests/todos_spec.rb]: We should updates specs accordingly, https://medium.com/just-tech/rspec-controller-or-request-specs-d93ef563ef11

File: features/todo.feature (hunk 1 above)
Comment (pdany1116) [reply features/todo.feature:10]: Find a solution to run all tests + rubocop.

File: app/controllers/todos_controller.rb (hunk 2 above)
Comment (pdany1116) [reply app/controllers/todos_controller.rb:13]: https://trello.com/c/7LCo514V/21-serialize-responses

File: features/todo.feature (hunk 1 above)
Comment (pdany1116) [reply features/todo.feature:10]: 4d40c5d3473a67df8c118e7ec46d95650ba2bd38

Comment (pdany1116) [reply features/todo.feature:10]: 4d40c5d3473a67df8c118e7ec46d95650ba2bd38

File: spec/requests/todos_spec.rb (hunk 4 above)
Comment (pdany1116) [reply spec/requests/todos_spec.rb]: 7545e194cc0b6a41e2d4b9e086c6a336b282e266

**Review by cazacutudor — APPROVED** [review]
:rocket:

**Pull request merged**